dspace-spring-rest/src/test/java/org/dspace/app/opensearch/OpenSearchControllerDisabledTest.java

```diff
@@ -0,0 +1,200 @@
+import org.dspace.app.rest.matcher.BrowseIndexMatcher;
+import org.dspace.app.rest.matcher.ItemMatcher;
+import org.dspace.app.util.service.OpenSearchService;
+import org.dspace.content.Collection;
+import org.dspace.content.Community;
+import org.dspace.content.Item;
+import org.dspace.eperson.Group;
+*/
+
+import org.dspace.app.rest.test.AbstractControllerIntegrationTest;
+
+import org.dspace.services.ConfigurationService;
+import org.dspace.services.factory.DSpaceServicesFactory;
+
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Integration test to test the /opensearch endpoint
+ * (Class has to start or end with IT to be picked up by the failsafe plugin)
+ *
+ * @author Oliver Goldschmidt (o dot goldschmidt at tuhh dot de)
+ */
+public class OpenSearchControllerDisabledTest extends AbstractControllerIntegrationTest {
+
+    private ConfigurationService configurationService;
+
+    @Before
+    public void init() throws Exception {
+        configurationService = DSpaceServicesFactory.getInstance().getConfigurationService();
+        configurationService.setProperty("websvc.opensearch.enable", false);
+/*
+System.out.println("Testing OpenSearch");
+MockHttpServletResponse resp2 = getClient().perform(get("/opensearch/search")
+                                .param("query", "dog"))
+                               .andReturn().getResponse();
+System.out.println("Response from Test 2: "+resp2.getContentAsString());
+*/
+    }
+
+    @Test
+    public void searchTest() throws Exception {
+        //When we call the root endpoint
+        getClient().perform(get("/opensearch/search")
+                                .param("query", "dog"))
+                   //The status has to be 200 OK
+                   .andExpect(status().isOk())
+                   //We expect the content type to be "application/hal+json;charset=UTF-8"
+                   .andExpect(content().contentType("text/html"))
+/*
+                   //Our default Discovery config has 4 browse indexes so we expect this to be reflected in the page
+                   // object
+                   .andExpect(jsonPath("$.page.size", is(20)))
+                   .andExpect(jsonPath("$.page.totalElements", is(4)))
+                   .andExpect(jsonPath("$.page.totalPages", is(1)))
+                   .andExpect(jsonPath("$.page.number", is(0)))
+
+                   //The array of browse index should have a size 4
+                   .andExpect(jsonPath("$._embedded.browses", hasSize(4)))
+
+                   //Check that all (and only) the default browse indexes are present
+                   .andExpect(jsonPath("$._embedded.browses", containsInAnyOrder(
+                       BrowseIndexMatcher.dateIssuedBrowseIndex("asc"),
+                       BrowseIndexMatcher.contributorBrowseIndex("asc"),
+                       BrowseIndexMatcher.titleBrowseIndex("asc"),
+                       BrowseIndexMatcher.subjectBrowseIndex("asc")
+                   )))
```

Comment:
More commented out tests. If these should remain commented out, it'd be good to add an explanation for when they should be enabled.

Reply:
removed comments